test/Microsoft.AspNetCore.SignalR.Client.Tests/HttpConnectionTests.cs

```diff
@@ -228,73 +463,37 @@ public void CannotStartConnectionWithInvalidTransportType(TransportType requeste
 
                     return request.Method == HttpMethod.Get
                         ? ResponseUtils.CreateResponse(HttpStatusCode.InternalServerError)
-                        : IsNegotiateRequest(request)
+                        : ResponseUtils.IsNegotiateRequest(request)
                             ? ResponseUtils.CreateResponse(HttpStatusCode.OK, ResponseUtils.CreateNegotiationResponse())
                             : ResponseUtils.CreateResponse(HttpStatusCode.OK);
                 });
 
             var connection = new HttpConnection(new Uri("http://fakeuri.org/"), TransportType.LongPolling, loggerFactory: null,
                 httpOptions: new HttpOptions { HttpMessageHandler = mockHttpHandler.Object });
+            var closeTcs = new TaskCompletionSource<object>();
+            connection.Closed += e =>
+            {
+                if (e != null)
+                {
+                    closeTcs.SetException(e);
+                }
+                else
+                {
+                    closeTcs.SetResult(null);
+                }
+            };
 
             try
             {
                 await connection.StartAsync().OrTimeout();
-                await Assert.ThrowsAsync<HttpRequestException>(() => connection.Closed.OrTimeout());
+                await Assert.ThrowsAsync<HttpRequestException>(() => closeTcs.Task.OrTimeout());
             }
             finally
             {
                 await connection.DisposeAsync();
             }
         }
 
-        [Fact]
-        public async Task ReceivedCallbackNotRaisedAfterConnectionIsDisposed()
```

Comment:
This is the test that I mentioned in other comments. It is not really relevant any more as the channel is closed before the state changes to make this event no-op occur.